Sofa/Component/SolidMechanics/Spring/src/sofa/component/solidmechanics/spring/RestShapeSpringsForceField.h

```diff
@@ -114,6 +114,11 @@ class RestShapeSpringsForceField : public core::behavior::ForceField<DataTypes>
     const VecIndex& getIndices() const { return m_indices; }
     const VecIndex& getExtIndices() const { return (useRestMState ? m_ext_indices : m_indices); }
 
+    static constexpr bool isRigid()
+    {
+        return std::string_view(DataTypes::Name()).find("Rigid") != std::string_view::npos;
```

Comment:
Is using string comparison the only way to make compile time type detection ?

Reply:
I was hesitating to test on the coord size (which is 4+3) but seems not really fail-proof IMO
I would like something to tell me this DataType is Rigid but I did not find.

Reply:
In any case, I suggest to introduce this function as a free global function in `sofa::helper`.

An alternative to comparing the name would be to use SFINAE or introduce a constexpr variable in all defaulttypes.

Reply:
> An alternative to comparing the name would be to use SFINAE 

That's what I wanted to avoid 🙊 but I guess I have no choice 👺
At least the test on Name() is much easier to read 🥲

 by the way, I saw it will be much easier with `requires` (c++20 😥)

> or introduce a constexpr variable in all defaulttypes.

I would prefer avoid that, I dont really like adding something for that kind of stuff. 
Otherwise if in the future we add a WhateverDataType, we will need to add it to all the other DataTypes `isWhatever` and so on.

> In any case, I suggest to introduce this function as a free global function in sofa::helper.

Could be an other solution, but not in sofa::helper, as it does not know stuff from sofa::defaulttype